meteor/client/ui/Settings/SystemManagement.tsx

```diff
@@ -282,3 +216,104 @@ export default translateWithTracker<IProps, {}, ITrackedProps>((_props: IProps)
 		}
 	}
 )
+
+export function checkForOldDataAndCleanUp(t: TFunction, retriesLeft: number = 0): void {
+	MeteorCall.system
+		.cleanupOldData(false)
+		.then((results) => {
+			console.log(results)
+			if (typeof results === 'string') {
+				if (retriesLeft <= 0) {
+					doModalDialog({
+						title: t('Error when checking for cleaning up'),
+						message: results,
+						acceptOnly: true,
+						onAccept: () => {
+							// nothing
+						},
+					})
+				} else {
+					// Try again:
+					Meteor.setTimeout(() => {
+						checkForOldDataAndCleanUp(t, retriesLeft - 1)
+					}, 300)
+				}
+			} else {
+				const collections = Object.values(results).filter((o) => o.docsToRemove > 0)
+				collections.sort((a, b) => {
+					return a.docsToRemove - b.docsToRemove
+				})
+
+				let totalCount = collections.reduce((memo: number, o) => {
+					return memo + o.docsToRemove
+				}, 0)
+
+				const affectedCollections: string[] = []
+				_.each(results, (result) => {
+					totalCount += result.docsToRemove
+					if (result.docsToRemove > 0) {
+						affectedCollections.push(result.collectionName)
+					}
+				})
+				if (totalCount) {
+					doModalDialog({
+						title: t('Remove old data from database'),
+						message: (
+							<React.Fragment>
+								<div>
+									{t('There are {{count}} documents that can be removed, do you want to continue?', {
+										count: totalCount,
+										collections: languageAnd(t, affectedCollections),
+									})}
+								</div>
+								<div>
+									{t('Documents to be removed:')}
+									{collections.map((o, index) => {
```

Comment:
This is a list, so it should be wrapped in a `<ul></ul>` and each item should be `<li key={index}></li>`